### PR TITLE
fix: Aria for tabs, card cluster, table sort

### PR DESCRIPTION
**Pull request by hitjim**
- screen reader reads card cluster selection and optionally also the longform
  description
- possible aria fix by adding optional  param for tablist, which allows use of <button> as tab element instead
  of <a>
- screen reader reads table sort type via adding proper enums, table element
  roles

Re-roll of #298 - build was broken, so we modified `main` branch history to not have a broken release.
- Also remembered to bump the minor version in `package.json`. 

Cause of prior breakage: Didn't change `href` to `onclick` on the new `<button>` tab element.

It also changed default behavior of tablist in a way that would force adoption of new, _possibly_ breaking functionality. So that was changed to be opt-in.

Here's a copy/paste of the previous PR description:
Previous implementation resulted in screen reader not reading when a tab was selected, or card cluster radio was selected, or table column was sorted.

Changes:

- Use `<button>` instead of `<a>` for tabs
- add `role="tab"`
- detect when `showRadioButton` is used, and apply `role="radio"` or `radiogroup` where applicable. If not, leave as `button`
- create and apply unique `id` for cards in `ts` file instead of repeatedly in the template.
 - enhance card cluster to allow using the `name` value (usually used like a description) as `aria-describedby`. This way the shorthand label is still read when describing the radio and its selection, followed by a more detailed description of that selection's purpose.
- tidy up table component's table-related `roles` where possible. Added `aria-sort`-related enums and attempted applying them in the template.

References for proper tab implementation, so as to comply with default screen reader / a11y behavior
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role
- https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML
- https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML#more_on_links
- https://css-tricks.com/use-button-element/#aa-accessibility-concerns

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bellese/angular-design-system",
-  "version": "4.3.0",
+  "version": "4.4.0",
```
Comment (mcelestini) [inline package.json:3]: did you make this change on purpose? I think the semantic versioning release bot will take care of this

File: src/app/modules/card-cluster/card-cluster.component.html [hunk 2]
```diff
@@ -52,18 +52,20 @@
...
+            id="{{ card.id }}"
+            [ngClass]="{ active: cardArray.buttonClicked === card.id, disabled: card.disabled }"
             class="{{ 'pointer card cardHeight ds-u-padding--1 ds-u-margin-bottom--2 ' + card.classButton }}"
             (click)="passAction($event)"
             (keyup.enter)="passAction($event)"
-            role="button"
+            [attr.role]="cardArray.showRadioButton ? 'radio' : 'button'"
             [attr.tabindex]="card.disabled ? null : 0"
             [attr.aria-label]="card.ariaLabel || card.name"
+            [attr.aria-describedby]="card.ariaDecribedbyName ? card.id + 'name' : undefined"
```
Comment (mcelestini) [inline src/app/modules/card-cluster/card-cluster.component.html:67]:
```suggestion
            [attr.aria-describedby]="card.ariaDescribedbyName ? card.id + 'name' : undefined"
```

File: src/app/modules/card-cluster/card-cluster.component.ts [hunk 3]
```diff
@@ -94,6 +94,20 @@ export class AppCardClusterComponent implements OnInit {
...
+      if (this.cardArray.clusterIdPrefix) {
+        idPrefix = this.cardArray.clusterIdPrefix;
+      }
+      this.cardArray.cluster.forEach(function(card, index) {
+        // Set unique IDs for each card, if not already explicitely defined.
+        if (!card.id) {
+          card.id = idPrefix ? `${idPrefix}${index}` : `CardCluster${index + 1}`;
+        }
+      });
+    }
```
Comment (mcelestini) [inline src/app/modules/card-cluster/card-cluster.component.ts:110]: what do you think about getting rid of this `idPrefix` variable? We could do it all by just checking `this.cardArray.clusterIdPrefix`? Although is a bit more of a mouthful 🤔 

```suggestion
    if (this.cardArray.cluster.length > 0) {
      this.cardArray.cluster.forEach(function(card, index) {
        // Set unique IDs for each card, if not already explicitely defined.
        if (!card.id) {
          card.id = this.cardArray.clusterIdPrefix ? `${this.cardArray.clusterIdPrefix}${index}` : `CardCluster${index + 1}`;
        }
      });
    }
```

File: src/app/modules/card-cluster/card-cluster.models.ts [hunk 4]
```diff
@@ -5,8 +5,10 @@ export class CardClusterClusterModel extends AngularDesignSystemModel {
   value: string | number;
   name?: string;
   nameIcon?: IconDefinition;
+  id?: string;
   valueIcon?: IconDefinition;
   ariaLabel?: string;
+  ariaDescribedbyName?: false;
```
Comment (mcelestini) [inline src/app/modules/card-cluster/card-cluster.models.ts:11]: nit: lowercase `by`
```suggestion
  ariaDescribedByName?: false;
```

File: src/app/modules/card-cluster/card-cluster.stories.ts [hunk 5]
```diff
@@ -90,19 +98,22 @@ const cardClusterDataShowRadioButtonsAndIcons = new CardClusterModel({
...
       ariaLabel: 'Activate enter to select Basic User',
+      ariaDecribedbyName: true,
       classButton: 'ds-u-padding--2',
     },
     {
       value: 'Security Administrator',
       name: 'A Security Administrator is a person who manages User Roles & Permissions for their Organization.',
       valueIcon: faUserShield,
       ariaLabel: 'Activate enter to select Security Administrator',
+      ariaDecribedbyName: true,
```
Comment (mcelestini) [inline src/app/modules/card-cluster/card-cluster.stories.ts:116]:
```suggestion
      ariaDescribedbyName: true,
```

File: src/app/modules/tabs/tabs.component.ts [hunk 6]
```diff
@@ -67,6 +73,7 @@ export class AppTabsComponent implements OnInit {
 
     }
 
+    // Deprecated?: This may be redundant and/or unused. Seems like most of the handling uses id via (click) and passAction.
```
Comment (mcelestini) [inline src/app/modules/tabs/tabs.component.ts:76]: can we remove this comment if we aren't sure if it's deprecated or not? Or if it is deprecated, can we remove the function 🧐

File: package.json (hunk 1 above)
Comment (hitjim) [reply package.json:3]: Confirmed, reverting to `4.3.0`

File: src/app/modules/card-cluster/card-cluster.component.html (hunk 2 above)
Comment (hitjim) [reply src/app/modules/card-cluster/card-cluster.component.html:67]: Fixed typo and camelCase. Added hyphen `'name'` -> `'-name'`

File: src/app/modules/card-cluster/card-cluster.component.ts (hunk 3 above)
Comment (hitjim) [reply src/app/modules/card-cluster/card-cluster.component.ts:110]: Accepting this in my next commit, since I had others I needed to do locally anyhow. 👍

File: src/app/modules/card-cluster/card-cluster.models.ts (hunk 4 above)
Comment (hitjim) [reply src/app/modules/card-cluster/card-cluster.models.ts:11]: 100% `camelCase` 👍

File: src/app/modules/card-cluster/card-cluster.stories.ts [hunk 7]
```diff
@@ -90,19 +98,22 @@ const cardClusterDataShowRadioButtonsAndIcons = new CardClusterModel({
...
   showRadioButton: true,
   rowMaxItems: 2,
+  clusterIdPrefix: 'userSelectCard',
   cluster: [
     {
       value: 'Basic User',
       name: 'A Basic User is an Organization Role with Read/Write Access to the Organization(s) in their system.',
       valueIcon: faEye,
       ariaLabel: 'Activate enter to select Basic User',
+      ariaDecribedbyName: true,
```
[the remark this answers is not in the document]
Comment (hitjim) [reply src/app/modules/card-cluster/card-cluster.stories.ts:108]: `ariaDescribedByName` ✅

File: src/app/modules/tabs/tabs.component.ts (hunk 6 above)
Comment (hitjim) [reply src/app/modules/tabs/tabs.component.ts:76]: removed

**Review by hitjim — COMMENTED** [review]
Didn't mean to wrap this as a review - but hey anyhow I followed up with all Marc's suggestions, via chat where necessary. Commit inbound.

File: src/app/modules/card-cluster/card-cluster.component.ts [hunk 8]
```diff
@@ -94,6 +94,16 @@ export class AppCardClusterComponent implements OnInit {
       }
     }
     this.resize();
+
+    // Handle optional parameters appropriately
+    if (this.cardArray.cluster.length > 0) {
+      this.cardArray.cluster.forEach(function(card, index) {
```
Comment (hitjim) [inline src/app/modules/card-cluster/card-cluster.component.ts:100]: Ahhh, that's right - I also needed to set `idPrefix` before entering the `forEach` b/c it's only on the parent object to the cluster array we're looping over. So running without I get a bunch of errors stating `cardArray` is undefined prop.

File: package-lock.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bellese/angular-design-system",
-  "version": "4.0.9",
+  "version": "4.4.0",
```
Comment (mcelestini) [inline package-lock.json:3]: could you also undo this change too if you don't mind? I think this was probably a downstream effect from running npm build with the same change in the `package.json` file

Comment (hitjim) [reply package-lock.json]: Done! Yeah missed that b/c of how GH collapses this file - which normally I wouldn't mind if the change wasn't a 1-liner 🤷

**Review by mcelestini — APPROVED** [review]
